plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -171,11 +177,31 @@ fu_thunderbolt_device_check_safe_mode (FuThunderboltDevice *self)
 	fu_device_set_metadata_boolean (FU_DEVICE (self), FU_DEVICE_METADATA_TBT_IS_SAFE_MODE, TRUE);
 }
 
+static const gchar*
+fu_thunderbolt_device_type_to_string (FuThunderboltDevice *self)
+{
+	if (self->device_type == FU_THUNDERBOLT_DEVICE_TYPE_HOST_CONTROLLER) {
+		if (self->gen >= 4)
+			return "USB4 host controller";
+		else
+			return "Thunderbolt host controller";
+	}
+	if (self->device_type == FU_THUNDERBOLT_DEVICE_TYPE_DEVICE_CONTROLLER) {
+		if (self->gen >= 4)
+			return "USB4 device controller";
+		else
+			return "Thunderbolt device controller";
+	}
+	if (self->device_type == FU_THUNDERBOLT_DEVICE_TYPE_RETIMER)
+		return "Retimer";
```

Comment:
I don't think the user is going to be familiar with that when it appears in `get-devices`...

Reply:
Do you have a suggestion?

Reply:
`USB4 extension device` perhaps?

Reply:
What about if it's renamed to USB4 retimer and put something along those lines in the summary?

Reply:
Yes, i think that's fine. Thanks.